apps/src/templates/GDPRDialog.jsx

```diff
@@ -0,0 +1,82 @@
+import React, {Component, PropTypes} from 'react';
+import Button from './Button';
+import i18n from "@cdo/locale";
+import BaseDialog from './BaseDialog';
+import DialogFooter from "./teacherDashboard/DialogFooter";
+import {pegasus} from '@cdo/apps/lib/util/urlHelpers';
+
+const styles = {
+  dialog: {
+    paddingLeft: 20,
+    paddingRight: 20,
+    paddingBottom: 20
+  },
+  instructions: {
+    marginTop: 20
+  }
+};
+
+export default class GDPRDialog extends Component {
+  static propTypes = {
+    isDialogOpen: PropTypes.bool.isRequired,
+    currentUserId: PropTypes.number.isRequired,
+    requestIp: PropTypes.string.isRequired,
+    // If we're coming from pegasus, studioUrlPrefix is passed in as a prop and used to construct the logout url.
+    studioUrlPrefix: PropTypes.string
+  };
+
+  state = {
+    isDialogOpen: this.props.isDialogOpen,
+  };
+
+  handleYesClick = () => {
+    this.setState({isDialogOpen: false});
+    $.post(
+      `/api/v1/users/${this.props.currentUserId}/accept_data_transfer_agreement`, {
+        data_transfer_agreement_request_ip: this.props.requestIp,
+        data_transfer_agreement_source: "DIALOG",
+        // TODO add in the agreement kind here!
+      }
+    );
+  };
+
+  render() {
+    const {studioUrlPrefix} = this.props;
+    const logOutUrl = studioUrlPrefix ?
+      `http:${studioUrlPrefix}/users/sign_out` : "/users/sign_out";
+
+    return (
+      <div>
+        <BaseDialog
+          useUpdatedStyles
+          isOpen={this.state.isDialogOpen}
+          style={styles.dialog}
+          uncloseable
+        >
+          <h2>{i18n.gdprDialogHeader()}</h2>
+          <div>
+            {i18n.gdprDialogDetails()}
+          </div>
+          <div style={styles.instructions}>
+            {i18n.gdprDialogLinkIntro() + " "}
```

Comment:
Again, connecting these two strings doesn't seem likely to work for other locales.

Reply:
I don't know of a better option in this case...

Reply:
We should at minimum be translating entire sentences at a time.  In this case it means an entire sentence will be a link, rather than just some words inside a sentence.

Reply:
Yes, we'll make the whole sentence a link (Poorva is OK with this change too). See here: https://github.com/code-dot-org/code-dot-org/pull/22541